src/CosmosEditorManager.ts

```diff
@@ -8,19 +8,32 @@ import * as os from 'os'
 import * as path from 'path';
 import * as vscode from 'vscode';
 import { DialogBoxResponses } from './constants';
-import { UserCancelledError } from 'vscode-azureextensionui';
+import { UserCancelledError, AzureTreeDataProvider, IAzureParentNode, IAzureNode } from 'vscode-azureextensionui';
 import * as util from './utils/vscodeUtils';
 import { randomUtils } from './utils/randomUtils';
+import { azureUtils } from './utils/azureUtils';
 import { MessageItem } from 'vscode';
+import { DocDBDocumentTreeItem } from './docdb/tree/DocDBDocumentTreeItem';
+import { DocDBDocumentNodeEditor } from './docdb/editors/DocDBDocumentNodeEditor';
+import { MongoDocumentTreeItem } from './mongo/tree/MongoDocumentTreeItem';
+import { MongoDocumentNodeEditor } from './mongo/editors/MongoDocumentNodeEditor';
+import { MongoCollectionTreeItem } from './mongo/tree/MongoCollectionTreeItem';
+import { MongoCollectionNodeEditor } from './mongo/editors/MongoCollectionNodeEditor';
 
 export interface ICosmosEditor<T = {}> {
     label: string;
+    id: EditableConfig;
```

Comment:
I've implemented some changes to the ui package to add more consistently formatted ids, with a `findNode` method that you can call directly. That way you should be able to persist just the id. Please take a look at this PR: https://github.com/Microsoft/vscode-azuretools/pull/76 (That PR is helpful for the functions extension, too - otherwise I might've had you do it yourself)